### PR TITLE
dojson: bd84188x field maps

### LINKED ISSUE #85: dojson: add mappings for bd84188x.py
- [ ] Add mappings (based on documentation from: https://www.loc.gov/marc/bibliographic/)
- [ ] Add test examples (for each mapping, take the example data from LOC website. For example, if you go to this page:
  https://www.loc.gov/marc/bibliographic/bd383.html
  you will see lines like:

```
383 ##$ano. 14,$bop. 27, no. 2
383 ##$cBWV 211
383 ##$bop. 8, no. 1-4
, etc.
```

The idea is to convert those lines into MARCXML example data (don't do this by hand, write a script for that !), so we are sure all mappings are tested:

```
<record>
  <datafield tag="383" ind1=" " ind2=" ">
    <subfield code="a">no. 14</subfield>
    <subfield code="b">op. 27, no. 2</subfield>
  </datafield>
  <datafield tag="383" ind1=" " ind2=" ">
    <subfield code="c">BWV 211</subfield>
  </datafield>
  <datafield tag="383" ind1=" " ind2=" ">
    <subfield code="b">op. 8, no. 1-4</subfield>
  </datafield>
, etc.
</record>
```

### PR DESCRIPTION
**Pull request by switowski**
- NEW Add more rules for MARC 21. (closes #85)

Signed-off-by: Sebastian Witowski witowski.sebastian@gmail.com

### CODE REVIEW

**Comment by switowski** [comment]
No more WiP, @SamiHiltunen can you take a look ?

File: dojson/contrib/marc21/fields/bd84188x.py [hunk 1]
```diff
@@ -14,12 +14,20 @@
 from ..model import marc21
 
 
-@marc21.over('holding_institution', '^850..')
+@marc21.over('holding_institution', '^850[_][_]')
```
Comment (SamiHiltunen) [inline dojson/contrib/marc21/fields/bd84188x.py]: this can be just __

File: dojson/contrib/marc21/fields/bd84188x.py [hunk 2]
```diff
@@ -29,13 +37,13 @@ def holding_institution(self, key, value):
...
 
 
-@marc21.over('location', '^852[_103254768][10_2]')
+@marc21.over('location', '^852[_12345678][_012]')
 @utils.for_each_value
 @utils.filter_values
 def location(self, key, value):
     """Location."""
     indicator_map1 = {
-        "#": "No information provided",
```
Comment (SamiHiltunen) [inline dojson/contrib/marc21/fields/bd84188x.py:38]: not a major thing but might be worth it to replace " with ' to make the code follow the conventions used elsewhere

File: dojson/contrib/to_marc21/fields/bd84188x.py [hunk 3]
```diff
@@ -300,20 +318,39 @@ def reverse_replacement_record_information(self, key, value):
...
+        'validity_end_date': 'x',
+        'uniform_resource_identifier': 'u',
+        'bibliographic_record_control_number': 'w',
+        'authority_record_control_number_or_standard_number': '0',
+        'field_link_and_sequence_number': '8',
+    }
+
+    order = utils.map_order(field_map, value)
+
+    if key[3] in indicator_map1:
```
Comment (SamiHiltunen) [inline dojson/contrib/to_marc21/fields/bd84188x.py]: this if statement can probably be removed

**Comment by SamiHiltunen** [comment]
Looks good, I would create test data for the missed fields and indicators to ensure all of the fields work.

**Comment by switowski** [comment]
Fixed. @egabancho can you, please, take a look ? Especially on line: https://github.com/inveniosoftware/dojson/pull/102/files#diff-4136e2cde53c12b5a3af8cc3394f138cR95 as I had no idea how to make it better (the name of the subfield has to be different, depending on the value of `ind1`)

**Comment by switowski** [comment]
@egabancho  :ping

**Comment by egabancho** [comment]
Remember to make a PR (or issue) on https://github.com/inveniosoftware/invenio-marc21 to update the schemas accordingly 

Should we put a `TODO` or some kind of warning somewhere for `880`? https://www.loc.gov/marc/bibliographic/bd880.html

**Comment by switowski** [comment]
There is an issue for 880 already: https://github.com/inveniosoftware/dojson/issues/101 (probably we skip it for now, if no one is using it) :wink: 
There is also an issue for the mismatched schemas: https://github.com/inveniosoftware/dojson/issues/99 that we still need to discuss (if I remember correctly, there was this idea of moving schemas out from dojson to a separate package - @tiborsimko can tell you more).

**Comment by SamiHiltunen** [comment]
Before merging, the commit message should be renamed to match the others eg. `dojson: bd84188x field maps` . Also removes the verb from the subject line ;)

**Comment by switowski** [comment]
@SamiHiltunen :+1: 
@egabancho it's merging time :hammer: !

**Pull request merged**